packages/cozy-realtime/src/CozyRealtime.js

```diff
@@ -1,360 +1,534 @@
-      this._socket.subscribe(type, id)
-    })
+  /**
+   * Register event handlers for cozy-client events
+   * @private
+   */
+  subscribeClientEvents() {
+    this.client.on('login', this.onClientLogin)
+    this.client.on('tokenRefreshed', this.onClientTokenRefreshed)
+    this.client.on('logout', this.onClientLogout)
   }
 
   /**
-   * Remove the given handler from the list of handlers for given
-   * doctype/document and event.
-   *
-   * @param {String}  type      Document doctype to unsubscribe from
-   * @param {String}  id        Document id to unsubscribe from
-   * @param {String}  eventName Event to unsubscribe from
-   * @param {Function}  handler   Function to call when an event of the
-   * given type on the given doctype or document is received from stack.
+   * Unregister event handlers for cozy-client events
+   * @private
    */
-  unsubscribe(eventName, type, handlerOrId, handlerOrUndefined) {
-    const { handler, id } = getHandlerAndId(handlerOrId, handlerOrUndefined)
-    validateParameters(eventName, type, id, handler)
-    const key = generateKey(eventName, type, id)
-
-    this.removeListener(key, handler)
+  unsubscribeClientEvents() {
+    this.client.removeListener('login', this.onClientLogin)
+    this.client.removeListener('tokenRefreshed', this.onClientTokenRefreshed)
+    this.client.removeListener('logout', this.onClientLogout)
+  }
 
-    if (!this._haveEventHandler()) {
-      this._resetSocket()
+  /**
+   * When the cozy-client instance successfully login - re-authenticate
+   * @private
+   */
+  onClientLogin() {
+    logger.info('Received login from cozy-client')
+    if (this.isWebSocketOpen()) {
+      this.authenticate()
+      // send subscriptions again, permissions may have changed
+      this.sendSubscriptions()
     }
   }
 
   /**
-   * Unsubscribe all handlers and close socket
+   * When the cozy-client instance refresh its access token - re-authenticate
+   * @private
    */
-  unsubscribeAll() {
-    this._getEventKeys()
-      .map(key => this._events[key].length > 0 && key)
-      .filter(Boolean)
-      .forEach(key => (this._events[key] = []))
+  onClientTokenRefreshed() {
+    // no need to authenticate again
+  }
 
-    this._resetSocket()
+  /**
+   * When the cozy-client instance logs out - throw the socket, don't reconnect yet (no authentication)
+   * @private
+   */
+  onClientLogout() {
+    if (this.hasWebSocket()) {
+      this.rejectWebSocket()
+    }
   }
 
-  async send(doctype, id, data) {
-    return this._cozyClient.stackClient.fetchJSON(
-      'POST',
-      `/realtime/${doctype}/${id}`,
-      {
-        data
-      }
-    )
+  /* * * * * * * * * * * * * * * */
+  /* * Global (browser) EVENTS * */
+  /* * * * * * * * * * * * * * * */
+
+  /**
+   * Subscribe to global events
+   * @private
+   */
+  subscribeGlobalEvents() {
+    if (isInBrowser && window.addEventListener) {
+      window.addEventListener('online', this.onOnline)
```

Comment:
Since in a cordova context, we need to install the cordova-plugin-network-information plugin to have access to these events, we should check if this plugin is installed and warn the dev if it's not. 

Also, we should add somewhere that we should not call realtime.start() / subscribe() before the `deviceReady` call from Cordova

Reply:
These checks were not in the previous version. To you know how I can check that we are in a cordova layer, that we have the correct plugin installed, and if deviceReady has fired or not?

Reply:
Oh I forgot to put the link... we do this kind of check here https://github.com/cozy/cozy-libs/blob/master/packages/cozy-device-helper/src/plugins.js#L7 

We can not really check if `deviceReady` has been already called or not. Adding a note to the readme should be enough IMHO.

Reply:
waiting for https://github.com/cozy/cozy-libs/pull/969

Reply:
For deviceready, I added an event handler for that in order to re-attach online/offline event handler if ever  the deviceready fires after cozy-realtime init.

Reply:
> For deviceready, I added an event handler for that in order to re-attach online/offline event handler if ever the deviceready fires after cozy-realtime init.

Thanks!